### PR TITLE
feat: [#1275] enhance live update

### LINKED ISSUE #1275: Enhance live update
- [ ] add descriptive logs
- [ ] enhance code
- [ ] function doc strings

### PR DESCRIPTION
**Pull request by wiktorflorian**
Closes #1275

### CODE REVIEW

File: transform/app/services/jms/connector.py [hunk 1]
```diff
@@ -46,14 +68,16 @@ def __init__(
         self._ssl = _ssl
 
     def on_message(self, frame):
-        logger.debug("Received a message '%s'", frame.body)
+        logger.debug("JMSMessageHandler: Received a message '%s'", frame.body)
```
Comment (Michal-Kolomanski) [inline transform/app/services/jms/connector.py:71]:
```
class_name = self.__class__.__name__
logger.debug("%s: Received a message '%s'", class_name, frame.body)
```

File: transform/app/services/jms/connector.py [hunk 2]
```diff
@@ -111,22 +162,33 @@ async def connect(
         subscribe(conn, host, port, username, password, topics, subscription_id, _ssl)
 
     except KeyboardInterrupt:
+        logger.warning("KeyboardInterrupt detected, disconnecting from JMS.")
         conn.disconnect()
-    except ConnectFailedException:
-        logger.error(f"Failed to connect to the queue: {host}:{port}")
+    except ConnectFailedException as e:
+        logger.error(f"Failed to connect to the JMS server: {host}:{port}, error: {e}")
```
Comment (Michal-Kolomanski) [inline transform/app/services/jms/connector.py:168]: JMS server -> JMS

File: transform/app/services/jms/connector.py [hunk 3]
```diff
@@ -111,22 +162,33 @@ async def connect(
...
+    """
     if not settings.STOMP_SUBSCRIPTION:
+        logger.info("STOMP subscription is disabled in settings.")
         return
 
     while True:
         try:
             logger.info(
-                f"Trying to subscribe to the queue: {settings.STOMP_HOST}:{settings.STOMP_PORT}, topics: {settings.STOMP_TOPICS}, ssl={settings.STOMP_SSL}"
+                f"Attempting to subscribe to the JMS queue: {settings.STOMP_HOST}:{settings.STOMP_PORT}, topics: {settings.STOMP_TOPICS}, ssl={settings.STOMP_SSL}"
```
Comment (Michal-Kolomanski) [inline transform/app/services/jms/connector.py:191]: JMS queue -> JMS

File: transform/app/services/jms/connector.py [hunk 4]
```diff
@@ -18,14 +18,36 @@
 
 conn: stomp.Connection | None = None
 
+# Constant
+RETRY_INTERVAL_SECONDS = 60
```
Comment (Michal-Kolomanski) [inline transform/app/services/jms/connector.py:22]: RETRY_INTERVAL_SECONDS -> INIT_JMS_RETRY_INTERVAL. # seconds

File: transform/app/services/solr/validate.py [hunk 5]
```diff
@@ -35,29 +42,27 @@ def check_document_exists(
...
+                    f"Failed to query Solr ({url=}). Status: {req.status_code}."
+                    f"Data type: {col_name}, Solr collection: {s_col_name}, ID: {data_id}"
                 )
-
         except ReqConnectionError as e:
             check_if_exists.append(False)
             logger.error(
-                f"Connection failed {url=}."
-                f"Querying Solr has failed. Data type={col_name}, solr_col={s_col_name}. Details: {e}"
+                f"Connection failed to {url=}."
```
Comment (Michal-Kolomanski) [inline transform/app/services/solr/validate.py:61]: Solr connection failed, {url=}. Data type={col_name}, solr_col={s_col_name}. Details: {e}

File: transform/app/transform/live_update/training_ig.py [hunk 6]
```diff
@@ -22,6 +30,29 @@ def process_message(frame) -> None:
     suspended = frame_body["suspended"]
     status = frame_body["status"]
 
+    collection, data, data_id = _extract_data_from_frame(raw_collection, frame_body)
+
+    logger.info(f"Received message, type: {raw_collection}, id: {data_id}")
```
Comment (Michal-Kolomanski) [inline transform/app/transform/live_update/training_ig.py:35]: Received message -> Started to process message, type: {raw_collection}, id: {data_id}

File: transform/app/transform/live_update/training_ig.py [hunk 7]
```diff
@@ -22,6 +30,29 @@ def process_message(frame) -> None:
...
+
+    if action == "create":
+        _handle_create_action(active, suspended, status, collection, data, data_id)
```
Comment (Michal-Kolomanski) [inline transform/app/transform/live_update/training_ig.py:38]: please remove _ from those functions. They don't need to be "private"

File: transform/app/transform/live_update/training_ig.py [hunk 8]
```diff
@@ -22,6 +30,29 @@ def process_message(frame) -> None:
     suspended = frame_body["suspended"]
     status = frame_body["status"]
 
+    collection, data, data_id = _extract_data_from_frame(raw_collection, frame_body)
```
Comment (Michal-Kolomanski) [inline transform/app/transform/live_update/training_ig.py:33]: same

File: transform/app/transform/live_update/training_ig.py [hunk 9]
```diff
@@ -35,30 +66,64 @@ def process_message(frame) -> None:
...
+        collection (str): The name of the collection.
+        data (dict): The data to be processed.
+        data_id (str): The ID of the data.
+    """
+    if (
+        active
+        and not suspended
+        and status in ["approved resource", "approved interoperability record"]
+    ):
+        logger.info(f"Creating {collection}, ID: {data_id}")
```
Comment (Michal-Kolomanski) [inline transform/app/transform/live_update/training_ig.py:89]: Creating {collection}, ID: {data_id} ->
Create action -  {collection}, ID: {data_id}

File: transform/app/transform/live_update/training_ig.py [hunk 10]
```diff
@@ -35,30 +66,64 @@ def process_message(frame) -> None:
...
+        collection (str): The name of the collection.
+        data (dict): The data to be processed.
+        data_id (str): The ID of the data.
+    """
+    if (
+        active
+        and not suspended
+        and status in ["approved resource", "approved interoperability record"]
+    ):
+        logger.info(f"Updating {collection}, ID: {data_id}")
```
Comment (Michal-Kolomanski) [inline transform/app/transform/live_update/training_ig.py:110]: Update action - ....

File: transform/app/transform/live_update/training_ig.py [hunk 11]
```diff
@@ -35,30 +66,64 @@ def process_message(frame) -> None:
...
+        transform_batch.delay(collection, data, full_update=False)
+    else:
         if check_document_exists(collection, data_id):
-            logger.info(f"Delete action - type: {raw_collection}, id: {data_id}")
+            logger.info(f"Data no longer valid. Deleting {collection}, ID: {data_id}")
```
Comment (Michal-Kolomanski) [inline transform/app/transform/live_update/training_ig.py:114]: Delete action - ...

File: transform/app/transform/live_update/training_ig.py [hunk 12]
```diff
@@ -35,30 +66,64 @@ def process_message(frame) -> None:
...
+    """
+    Handles the 'delete' action for the JMS message.
+
+    Args:
+        collection (str): The name of the collection.
+        data_id (str): The ID of the data to be deleted.
+        data (dict): The data to be deleted.
+    """
+    if check_document_exists(collection, data_id):
+        logger.info(f"Deleting {collection}, ID: {data_id}")
```
Comment (Michal-Kolomanski) [inline transform/app/transform/live_update/training_ig.py:128]: Delete action - ...

**Review by Michal-Kolomanski — APPROVED** [review]

File: transform/app/services/jms/connector.py (hunk 1 above)
Comment (wiktorflorian) [reply transform/app/services/jms/connector.py:71]: done

File: transform/app/services/solr/validate.py (hunk 5 above)
Comment (wiktorflorian) [reply transform/app/services/solr/validate.py:61]: You are right, it makes it easier to read.

**Pull request merged**